### PR TITLE
Lets you chug from ALL the drinking containers 

### PR DESCRIPTION
**Pull request by TTerc**
[feat]

## About the PR 

This PR simply moves the clickdrag chugging proc from /obj/item/reagent_containers/food/drinks/drinkingglass to /obj/item/reagent_containers/food/drinks and modifies the actionbar a little bit, letting you chug from all the drinking containers.

## Why's this needed? 

You ever wished you could chug the entire cup of Discount Dan's Noodles? Now you can.

## Changelog

```changelog
(u)TTerc
(+)You can now chug from all the drinking containers, not just glasses and pitchers (with a few exceptions).
```

### CODE REVIEW

**Review by Tarmunora — APPROVED** [review]

**Comment by millicentjwi** [comment]
YES FUCK YES

**Comment by Tarmunora** [comment]
doublecheck that closed cans etc still work as expected ie, cannot chug a sealed can

**Comment by TTerc** [comment]
Good point. Fixed, no more chugging from sealed cans

**Comment by Tarmunora** [comment]
Hm. Consider a `can_chug` var on drinks instead, and set/unset as needed. Probably don't want to allow chugging from juice-o-matics, cocktail shakers(!), and maybe also disallow some stuff like water-coolers, fueltanks, and coffee carafes

**Comment by TTerc** [comment]
yeah, artpitchers aswell perhaps

**Comment by TTerc** [comment]
Cocktail shakers, juice-o-matics, artpitchers, fuel tanks, water cooler bottles and coffee carafes are now blacklisted

File: code/modules/food_and_drink/drinks.dm [hunk 1]
```diff
@@ -259,6 +259,12 @@
 			return
 		..()
 
+	MouseDrop(mob/user)
+		if (is_sealed)
+			boutput(user, "<span class='alert'>You can't drink out of a sealed can!</span>")
+			return
+		..()
+
```
Comment (Tarmunora) [inline code/modules/food_and_drink/drinks.dm:267]: use `can_chug` instead (default false, set to true in `attack_self()`)

File: code/modules/chemistry/tools/food_and_drink.dm [hunk 2]
```diff
@@ -460,6 +461,37 @@
...
+		..()
+		if(!(usr == over_object)) return
+		if(!istype(usr, /mob/living/carbon)) return
+		var/mob/living/carbon/C = usr
+
+		var/maybe_too_clumsy = FALSE
+		var/maybe_too_tipsy = FALSE
+		var/too_drunk = FALSE
+		if(!can_chug)
+			boutput(C, "<span class='alert'>You try to lift the [src.name] and attempt to chug from it but it's too heavy!</span>")
```
Comment (Tarmunora) [inline code/modules/chemistry/tools/food_and_drink.dm:474]:
```suggestion
			boutput(C, "<span class='alert'>You can't seem to chug from [src.name]! How odd.</span>")
```
cocktail shakers etc probably aren't that much heavier than a full pitcher 🤷

**Review by Tarmunora — CHANGES_REQUESTED** [review]

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**